src/Rating.jsx

```diff
@@ -22,11 +22,12 @@ class Rating extends React.PureComponent {
   }
 
   componentWillReceiveProps(nextProps) {
+    const valueChanged = this.props.value !== nextProps.value;
     this.setState({
-      dirty: this.props.value !== nextProps.value && !this.state.dirty
+      dirty: valueChanged && !this.state.dirty
```

Comment:
Not part of this pull request... But... The `dirty` state expression looks convoluted to me. Isn't it equivalent to:

```javascript
dirty: valueChanged || this.state.dirty
```

Reply:
Even more... Is it used somewhere?

```bash
$ grep dirty Rating.jsx
      dirty: false
      dirty: this.props.value !== nextProps.value && !this.state.dirty
        : this.state.dirty,
```

Reply:
It was also confusing to me. I didn't find any usage of it.